lib/dry/logic/predicates.rb

```diff
@@ -254,8 +254,13 @@ def uri_rfc3986?(input)
           format?(URI::RFC3986_Parser::RFC3986_URI, input)
         end
 
-        def respond_to?(method, input)
-          input.respond_to?(method)
+        def respond_to?(*args)
```

Comment:
`*args` allocated an array, I think you can achieve the same semantics with `respond_to?(m, input = Undefined)`. I prefer to avoid using `*` and `**` when possible for this reason.